src/extension/extension.ts

```diff
@@ -271,7 +272,25 @@ export async function activate(context: vs.ExtensionContext, isRestart: boolean
 			};
 		}
 
-		flutterDaemon = new FlutterDaemon(logger, workspaceContext as FlutterWorkspaceContext, flutterCapabilities, runIfNoDevices);
+		if (workspaceContext.config.forceFlutterWorkspace) {
+			let resultFromLocalExtension = null;
+			try {
+				resultFromLocalExtension = await vs.commands.executeCommand<string>("dartlocaldevice.startDaemon", workspaceContext.config.flutterToolsScript?.script, "expose_devices", workspaceContext.config.flutterSdkHome);
+			} catch (e) {
+				// Command may not be available.
+			}
+			if (resultFromLocalExtension !== null) {
+				const resultMessage = resultFromLocalExtension.toString();
+				const results = resultMessage.match(/Device daemon is available on remote port: (\d+)/i);
+				if (results !== null && results?.length > 1) {
+					portFromLocalExtension = parseInt(results[1]);
+				} else if (resultMessage !== null) {
+					vs.window.showErrorMessage(resultMessage);
```

Comment:
Generally I try to always write to the logger (`logger.error()`) when we show errors. Sometimes users don't notice the toast notifications and VS Code hides them after some number of seconds (5-10?). By writing them to the logger they'll show up both in the dev console (Help -> Toggle Developer Tools) and also if they're logging (with the Capture Logs command).